src/audio/component.c

```diff
@@ -496,6 +497,62 @@ void audio_stream_copy_to_linear(const struct audio_stream *source, int ioffset,
 	}
 }
 
+static inline bool comp_check_eos(struct comp_dev *dev)
```

Comment:
more readable since we are returning a bool as ```comp_is_eos()```

Reply:
This function not only checks if eos has occurred, but also handles it by setting the appropriate state on the sinks and in the case of dp modules, feeds them with silence. That's why I think the `comp_is_eos` name is not appropriate.